.github/workflows/build-and-test-arm.yml

```diff
@@ -0,0 +1,64 @@
+name: build and test on armv7 and aarch64 with QEMU
+on: [push, pull_request]
+jobs:
+  test-on-ubuntu-latest-with-qemu:
+    strategy:
+      matrix:
+        # only test specific parameter pairs, not the cross product
+        include:
+          - target: "armv7-unknown-linux-gnueabihf"
+            linker: "arm-linux-gnueabihf-gcc"
+            interpreter: "qemu-arm"
+            packages: "g++-arm-linux-gnueabihf libc6-dev-armhf-cross"
+          - target: "aarch64-unknown-linux-gnu"
+            linker: "aarch64-linux-gnu-gcc"
+            interpreter: "qemu-aarch64"
+            packages: "g++-aarch64-linux-gnu libc6-dev-arm64-cross"
+    runs-on: ubuntu-latest
+    name: test on ${{ matrix.target }}
+    steps:
+      - name: install prerequisites
+        uses: awalsh128/cache-apt-pkgs-action@latest
+        with:
+          packages: ${{ matrix.packages }} binfmt-support qemu-user-static meson nasm
+          version: 1.0 # version of cache to load
+        # since binfmt-support was likely cached from previous run, we need to
+        # re-run the post installation step from that package to register
+        # qemu-arm, qemu-aarch64, etc. as an interpreter for the cross compiled
+        # binaries.
+      - name: register ${{ matrix.interpreter }} as interpreter
+        run: |
+          sudo update-binfmts --import
+          sudo update-binfmts --enable ${{ matrix.interpreter }}
+          update-binfmts --display
+      - name: git checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+      - name: cache rust dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ matrix.target }}-cargo-and-target-${{ hashFiles('**/Cargo.lock') }}
+      - name: cargo build for ${{ matrix.target }}
+        run: |
+          rustup target add ${{ matrix.target }}
+          RUSTFLAGS="-C target-feature=+crt-static -C linker=${{ matrix.linker }}" \
+            cargo build --release --target ${{ matrix.target }}
```

Comment:
I'm not sure if we were doing this before, but we should be testing with checked release builds.  Something like this:

```toml
# Fast but checked.
# For CI and sometimes local development with qemu.
#
# That is, it compiles with all runtime checks and limited debug info (as opposed to none).
[profile.checked-release]
inherits = "release"
opt-level = 3
debug = "limited"
debug-assertions = true
overflow-checks = true
lto = "thin"
codegen-units = 16
```